src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationCollectionBindingTests.cs

```diff
@@ -393,6 +393,11 @@ public void AlreadyInitializedListInterfaceBinding()
             Assert.Equal("val1", list[2]);
             Assert.Equal("val2", list[3]);
             Assert.Equal("valx", list[4]);
+
+            // Ensure expandability of the returned list
+            options.AlreadyInitializedListInterface.Add("ExtraItem");
```

Comment:
Do we have a test for an `IList` or `ICollection` property, where the backing field hasn't been set yet?  Basically a test for the `if (bindingPoint.Value is null)` check.

Reply:
This test case already using `UninitializedCollectionsOptions` which include the uninitialized `ICollection`.  The extra added code to this test code is testing the returned collection to allow adding more items to it. 
The next test case `CanBindUninitializedIList` is test `IList` and I did the same to ensure the returned list allows adding more items.